File: PHP/Day 2 - Dive! Part One.php
<?php
/**
 * Created by PhpStorm.
 * User: angelgrigorov
 * Date: 02.12.21
 * Time: 10:01
 */
$fh = fopen('Day 2 - Input.txt','r');
$position = 0;
$depth = 0;
while ($line = fgets($fh)) {
    list($command, $number) = explode(" ",$line);
    if($command == "forward") {
        $position += intval($number);
    }elseif($command == "down") {
        $depth += intval($number);
    }elseif ($command == "up") {
        $depth -= intval($number);
    }
}
fclose($fh);

$result = $position * $depth;
echo $result;

File: GoLang/Day 2 - Dive! Part One.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	file, _ := os.Open("Day 2 - Input.txt")
	defer file.Close()
	scanner := bufio.NewScanner(file)
	position := 0
	depth := 0

	for scanner.Scan() {
		command := strings.Split(scanner.Text(), " ")
		intnumber, _ := strconv.Atoi(command[1])
		if command[0] == "forward" {
			position += intnumber
		} else if command[0] == "down" {
			depth += intnumber
		} else if command[0] == "up" {
			depth -= intnumber
		}

	}

	fmt.Println(position * depth)
}
